packages/vitest/src/integrations/chai/jest-expect.ts

```diff
@@ -579,6 +589,10 @@ export const JestChaiExpect: ChaiPlugin = (chai, utils) => {
     utils.flag(this, 'error', new Error('rejects'))
     const obj = utils.flag(this, 'object')
     const wrapper = typeof obj === 'function' ? obj() : obj // for jest compat
+
+    if (typeof wrapper?.then !== 'function')
+      throw new TypeError(`You must provide a Promise to expect() when using .rejects, not '${typeof wrapper}'.`)
```

Comment:
Couldn't we do a `Promise.resolve` to normalize it?

Reply:
I think it’s better to have strict types, otherwise people test code that Vitest creates for them instead of the actual code.